lib/gooddata/rest/connection.rb

```diff
@@ -466,6 +466,17 @@ def generate_request_id
         "#{@execution_id}:#{session_id}:#{call_id}"
       end
 
+      def enrich_error_message(exception)
+        begin
+          response = JSON.parse(exception.response.body, symbolize_names: true)
+          exception.define_singleton_method :to_s do
```

Comment:
Why monkey patch `.to_s`? It would be better to throw a new exception and retain the original exception using `Exception#cause` (happens automatically). Monkey patching is problematic. Moreover, `exception.message` says something else than `exception.to_s`.